app/models/rbac.rb

```diff
@@ -571,6 +571,10 @@ def self.get_belongsto_matches(blist, klass)
         end
       elsif klass == Host
         results.concat(get_belongsto_matches_for_host(vcmeta_list.last))
+      elsif vcmeta_list.last.kind_of?(Host) && klass <= VmOrTemplate
+        host = vcmeta_list.last
+        vms_and_templates = host.send(klass.to_s.tableize).to_a
```

Comment:
Is klass only ever Vm or MiqTemplate within this leg of the condition?  Is it ever `ManageIQ::Providers::Vmware::InfraManager::Vm` or other subclasses for other providers?  Host only has vms,   vms_and_templates, vms, miq_templates...